source/test/TestApp/AutoGenTests/AutoGenDataTestData.cpp

```diff
@@ -3,144 +3,270 @@
 #include "TestApp.h"
 #include "AutoGenDataTests.h"
 #include "XAsyncHelper.h"
-#include <playfab/PlayFabClientAuthApi.h>
-#include <playfab/PlayFabClientApi.h>
-#include <playfab/PlayFabProfilesApi.h>
-#include <playfab/PlayFabAdminApi.h>
-#include <playfab/PlayFabAuthenticationAuthApi.h>
-#include <playfab/PlayFabClientDataModels.h>
 
 namespace PlayFabUnit
 {
 
-void AutoGenDataTests::FillPlayFabDataAbortFileUploadsRequest( PlayFab::DataModels::AbortFileUploadsRequest* request )
-{
-    // TODO: debug  test
-    PlayFab::JsonDocument inputJson;
-    inputJson.Parse( "{ \"FileNames\": [ \"avatar\" ], \"Entity\": { \"Id\": \"ID\", \"Type\": \"title_player_account\", \"TypeString\": \"title_player_account\" }}" );
-    request->FromJson(inputJson);
-}
+#pragma region AbortFileUploads
 
-HRESULT AutoGenDataTests::ValidatePlayFabDataAbortFileUploadsResponse( PlayFabDataAbortFileUploadsResponse* result )
-{
-    // result.entity = PlayFabEntityKey const*;
-    // result.profileVersion = int32_t;
+    void AutoGenDataTests::FillAbortFileUploadsPrerequisiteInitiateFileUploadsRequest(PlayFab::DataModels::InitiateFileUploadsRequest* request)
```

Comment:
The implementation of this method and a few others appears to be identical. Would it be possible to bring in more code-reuse. Can each of these call a base implementation instead?

Reply:
Since these are all auto generated, I'm not sure if that would change much output-wise. Maybe I'm just missing it right now, but since they're all pretty tiny and all rely on bespoke data, I don't think it would cut down on the code significantly? Someone feel free to correct me if they see something here I don't though

Reply:
Understood on the auto-gen nature of these, but these are the templates, not the output. I'm slightly worried about future maintenance of these. If we want to update the initialization, validation, or clean-up logic, we have to do it across a bunch of places (that will continue to grow as we add more tests). I'm not pushing hard, though, as we can always refactor later if that becomes a problem.

Reply:
These are actually the output files generated from the TestData.cpp.ejs template. Since there's annoying manual data input for these, we have to generate the files and then copy them back here to have the actual data properly stored. At least that's how it was created I think. There may be some refactor we could do to move all the data to a separate json file that's read at test runtime, and then the actual structure of the generated TestData files can be changed willy nilly and they won't have to be stored here anymore...maybe? That'll be a relatively big refactor, but I'm sure it could be done between feature group work.

Reply:
Oh, you're right, I misread and thought this was an ejs file. Sorry! I'm fine with this staying.